File: eunseong/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    bool answer = true;
    int sum =0;
    int num = x;
    while(num >= 1){
        sum += (num%10);
        num = num/10;
    }
    
    answer = (x%sum == 0) ? true : false;
    return answer;
}

File: bsw112/2021-01-28/하샤드수.java
public class 하샤드수 {
    public boolean solution(int x) {
        String num = Integer.toString(x);
        int sum = 0;
        for (int i = 0; i < num.length(); i++) {
            sum += Integer.parseInt(Character.toString(num.charAt(i)));
        }
        return x % sum == 0;
    }
}
